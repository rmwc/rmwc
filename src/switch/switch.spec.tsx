import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Switch } from './';

describe('Switch', () => {
  test('renders', () => {
    const { asFragment } = render(<Switch />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('can be checked', () => {
    render(<Switch checked onChange={() => {}} />);
    expect(screen.getByRole('switch')).toHaveAttribute('checked');
  });

  test('handles onChange', async () => {
    let value = 0;
    render(<Switch checked={true} onChange={() => value++} />);

    userEvent.click(screen.getByRole('switch'));

    await waitFor(() => expect(value).toEqual(1));
  });

  test('can be disabled', () => {
    render(<Switch disabled />);
    expect(screen.getByRole('switch')).toHaveAttribute('disabled');
  });

  test('can have a label', () => {
    render(<Switch label="hello world" />);
    expect(screen.getByText('hello world')).toBeInTheDocument();
  });

  test('can have custom classnames on input', () => {
    const { container } = render(<Switch className={'my-custom-classname'} />);
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });

  it('supports inputRef as an object reference', () => {
    const inputObjectRef: any = { current: null };
    render(<Switch inputRef={inputObjectRef} />);
    expect(inputObjectRef.current instanceof HTMLInputElement).toBeTruthy();
  });

  it('supports inputRef as a function reference', () => {
    let inputObjectRef: any;
    const objectRefFunc: any = (el: HTMLInputElement) => {
      inputObjectRef = el;
    };
    render(<Switch inputRef={objectRefFunc} />);
    expect(inputObjectRef instanceof HTMLInputElement).toBeTruthy();
  });
});
