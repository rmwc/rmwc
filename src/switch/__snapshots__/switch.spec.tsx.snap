// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch renders 1`] = `
<DocumentFragment>
  <div
    class="mdc-switch"
  >
    <div
      class="mdc-switch__track"
    />
    <div
      class="mdc-ripple-upgraded--unbounded mdc-switch__thumb-underlay"
      data-mdc-ripple-is-unbounded="true"
    >
      <div
        class="mdc-switch__thumb"
      />
      <input
        class="mdc-switch__native-control"
        id="toggle--test"
        role="switch"
        type="checkbox"
      />
    </div>
    <div
      class="mdc-switch__knob"
    />
  </div>
</DocumentFragment>
`;
